.github/workflows/golangci-lint.yml

```diff
@@ -21,9 +21,9 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
-          go-version: 1.17
+          go-version: 1.19
```

Comment:
I would prefer staying on Go 1.18 for the moment until we upgrade everything else to the next Kubernetes and Go version. (applies to all occurences of Go 1.19 on this PR)

Otherwise the linter in CI would be the only thing using Go 1.19 while every other job and we locally would use Go 1.18

@killianmuldoon WDYT?

Reply:
From the release notes for 1.19:

```
Changes to the language

There is only one small change to the language, a [very small correction](https://github.com/golang/go/issues/52038) to the [scope of type parameters in method declarations](https://tip.golang.org/ref/spec#Declarations_and_scope). Existing programs are unaffected.
```
source: https://tip.golang.org/doc/go1.19

Reply:
My interpretation is that this only affects generic functions and since generics isn't used yet it should be fine.

Reply:
But it still feels wrong to me if we run golangci-lint in GitHub actions with Go 1.19 while the everything else uses Go 1.18.

E.g. when I run the linter on my machine I will use Go 1.18 as we use 1.18 for everything

Reply:
No problem then! Fixed.

Reply:
Thx!